File: mobile_apps/Android/deliveryman/dropoprovider/src/main/java/com/dropo/provider/models/datamodels/OrderTotal.java
package com.dropo.provider.models.datamodels;

import com.google.gson.annotations.SerializedName;

public class OrderTotal {
    @SerializedName("is_weekly_invoice_paid")
    private boolean isWeeklyInvoicePaid;

    @SerializedName("total_earning")
    private double totalEarning = 0;

    @SerializedName("provider_have_cash_payment")
    private double providerHaveCashPayment = 0;

    @SerializedName("wallet")
    private double wallet = 0;

    @SerializedName("total_provider_profit")
    private double totalProviderProfit = 0;

    @SerializedName("total_service_price")
    private double totalServicePrice = 0;

    @SerializedName("provider_paid_order_payment")
    private double providerPaidOrderPayment = 0;

    @SerializedName("total_admin_profit_on_delivery")
    private double totalAdminProfitOnDelivery = 0;

    @SerializedName("pay_to_provider")
    private double payToProvider = 0;

    @SerializedName("_id")
    private Object id;

    @SerializedName("total_admin_tax_price")
    private double totalAfterTaxPrice = 0;

    @SerializedName("total_surge_price")
    private double totalSurgePrice = 0;

    @SerializedName("total_delivery_price")
    private double totalDeliveryPrice = 0;
    @SerializedName("total_deduct_wallet_income")
    private double totalWalletIncomeSetInCashOrder = 0;
    @SerializedName("total_added_wallet_income")
    private double totalWalletIncomeSetInOtherOrder = 0;
    @SerializedName("total_provider_have_cash_payment_on_hand")
    private double totalProviderHaveCashPaymentOnHand = 0;
    @SerializedName("total_transferred_amount")
    private double totalPaid = 0;
    @SerializedName("total_wallet_income_set")
    private double totalWalletIncomeSet = 0;

    public double getTotalEarning() {
        return totalEarning;
    }

    public void setTotalEarning(double totalEarning) {
        this.totalEarning = totalEarning;
    }

    public double getProviderHaveCashPayment() {
        return providerHaveCashPayment;
    }

    public void setProviderHaveCashPayment(double providerHaveCashPayment) {
        this.providerHaveCashPayment = providerHaveCashPayment;
    }

    public double getWallet() {
        return wallet;
    }

    public void setWallet(double wallet) {
        this.wallet = wallet;
    }

    public double getTotalProviderProfit() {
        return totalProviderProfit;
    }

    public void setTotalProviderProfit(double totalProviderProfit) {
        this.totalProviderProfit = totalProviderProfit;
    }

    public double getTotalServicePrice() {
        return totalServicePrice;
    }

    public void setTotalServicePrice(double totalServicePrice) {
        this.totalServicePrice = totalServicePrice;
    }

    public double getProviderPaidOrderPayment() {
        return providerPaidOrderPayment;
    }

    public void setProviderPaidOrderPayment(double providerPaidOrderPayment) {
        this.providerPaidOrderPayment = providerPaidOrderPayment;
    }

    public double getTotalAdminProfitOnDelivery() {
        return totalAdminProfitOnDelivery;
    }

    public void setTotalAdminProfitOnDelivery(double totalAdminProfitOnDelivery) {
        this.totalAdminProfitOnDelivery = totalAdminProfitOnDelivery;
    }

    public double getPayToProvider() {
        return payToProvider;
    }

    public void setPayToProvider(double payToProvider) {
        this.payToProvider = payToProvider;
    }

    public Object getId() {
        return id;
    }

    public void setId(Object id) {
        this.id = id;
    }

    public double getTotalAfterTaxPrice() {
        return totalAfterTaxPrice;
    }

    public void setTotalAfterTaxPrice(double totalAfterTaxPrice) {
        this.totalAfterTaxPrice = totalAfterTaxPrice;
    }

    public double getTotalSurgePrice() {
        return totalSurgePrice;
    }

    public void setTotalSurgePrice(double totalSurgePrice) {
        this.totalSurgePrice = totalSurgePrice;
    }

    public double getTotalDeliveryPrice() {
        return totalDeliveryPrice;
    }

    public void setTotalDeliveryPrice(double totalDeliveryPrice) {
        this.totalDeliveryPrice = totalDeliveryPrice;
    }

    public boolean isWeeklyInvoicePaid() {
        return isWeeklyInvoicePaid;
    }

    public void setWeeklyInvoicePaid(boolean weeklyInvoicePaid) {
        isWeeklyInvoicePaid = weeklyInvoicePaid;
    }

    public double getTotalWalletIncomeSetInCashOrder() {
        return totalWalletIncomeSetInCashOrder;
    }

    public void setTotalWalletIncomeSetInCashOrder(double totalWalletIncomeSetInCashOrder) {
        this.totalWalletIncomeSetInCashOrder = totalWalletIncomeSetInCashOrder;
    }

    public double getTotalWalletIncomeSetInOtherOrder() {
        return totalWalletIncomeSetInOtherOrder;
    }

    public void setTotalWalletIncomeSetInOtherOrder(double totalWalletIncomeSetInOtherOrder) {
        this.totalWalletIncomeSetInOtherOrder = totalWalletIncomeSetInOtherOrder;
    }

    public double getTotalProviderHaveCashPaymentOnHand() {
        return totalProviderHaveCashPaymentOnHand;
    }

    public void setTotalProviderHaveCashPaymentOnHand(double totalProviderHaveCashPaymentOnHand) {
        this.totalProviderHaveCashPaymentOnHand = totalProviderHaveCashPaymentOnHand;
    }

    public double getTotalPaid() {
        return totalPaid;
    }

    public void setTotalPaid(double totalPaid) {
        this.totalPaid = totalPaid;
    }


    public double getTotalWalletIncomeSet() {
        return totalWalletIncomeSet;
    }

    public void setTotalWalletIncomeSet(double totalWalletIncomeSet) {
        this.totalWalletIncomeSet = totalWalletIncomeSet;
    }
}

File: mobile_apps/IOS/deliveryman/Edelivery Provider/Models/DataModel/Earning Response/OrderTotal.swift
//
//	OrderTotal.swift
//
//	Create by Elluminati on 28/6/2017
//	Copyright © 2017 Elluminati. All rights reserved.


import Foundation

class OrderTotal{

	var id : Any!
	var payToProvider : Double! = 0.0
	var providerHaveCashPayment : Double! = 0.0
	var providerPaidOrderPayment : Double! = 0.0
	var totalAdminProfitOnDelivery : Double! = 0.0
	var totalAdminTaxPrice : Double! = 0.0
	var totalDeliveryPrice : Double! = 0.0
	var totalEarning : Double! = 0.0
	var totalProviderProfit : Double! = 0.0
	var totalServicePrice : Double! = 0.0
	var totalSurgePrice : Double! = 0.0
	var totalProviderHaveCashPaymentOnHand : Double! = 0.0
    var totalWalletIncomeSet : Double! = 0.0
    var totalWalletIncomeSetInCashOrder : Double! = 0.0
    var totalWalletIncomeSetInOtherOrder : Double! = 0.0
    var totalRemainingToPaid : Double! = 0.0
    var totalPaid : Double! = 0.0
	/**
	 * Instantiate the instance using the passed dictionary values to set the properties values
	 */
	init(fromDictionary dictionary: [String:Any]){
		id = dictionary["_id"] ?? "0.0."
		payToProvider = (dictionary["pay_to_provider"] as? Double) ?? 0.0
		providerHaveCashPayment = (dictionary["provider_have_cash_payment"] as? Double) ?? 0.0
		providerPaidOrderPayment = (dictionary["provider_paid_order_payment"] as? Double) ?? 0.0
		totalAdminProfitOnDelivery = (dictionary["total_admin_profit_on_delivery"] as? Double) ?? 0.0
		totalAdminTaxPrice = (dictionary["total_admin_tax_price"] as? Double) ?? 0.0
		totalDeliveryPrice = (dictionary["total_delivery_price"] as? Double) ?? 0.0
		totalEarning = (dictionary["total_earning"] as? Double) ?? 0.0
		totalProviderProfit = (dictionary["total_provider_profit"] as? Double) ?? 0.0
		totalServicePrice = (dictionary["total_service_price"] as? Double) ?? 0.0
		totalSurgePrice = (dictionary["total_surge_price"] as? Double) ?? 0.0
        totalWalletIncomeSetInCashOrder = (dictionary["total_deduct_wallet_income"] as? Double) ?? 0.0
        totalWalletIncomeSetInOtherOrder = (dictionary["total_added_wallet_income"] as? Double) ?? 0.0
       
        totalProviderHaveCashPaymentOnHand = (dictionary["total_provider_have_cash_payment_on_hand"] as? Double) ?? 0.0
        
        totalPaid = (dictionary["total_transferred_amount"] as? Double) ?? 0.0
        totalRemainingToPaid = (dictionary["total_remaining_to_paid"] as? Double) ?? 0.0
		totalWalletIncomeSet = (dictionary["total_wallet_income_set"] as? Double) ?? 0.0
	}

}
